src/components/CartPage/CartPage.js

```diff
@@ -0,0 +1,138 @@
+
+import './CartPage.scss';
+
+const CN = 'cart-page';
+
+const CartPage = (props) => {
+  const {
+    className, incrementQuantity, decrementQuantity, removeItemFromCart, userCart
+  } = props;
+  const { cartTableHeadings } = config;
+  const products = [
+    {
+      id: 10,
+      title: 'beautiful swing coat',
+      image: 'https://static.massimodutti.net/3/photos/2019/I/0/1/p/6406/512/800/6406512800_2_2_16.jpg?t=1571072186173&impolicy=massimodutti-itxmediumhigh&imwidth=900',
+      color: 'blue',
+      size: 12,
+      price: 40,
+      inventory: 10
+    },
+    {
+      id: 12,
+      title: 'nice swingy jacket',
+      image: 'https://static.massimodutti.net/3/photos/2019/I/0/1/p/6747/710/808/6747710808_1_1_16.jpg?t=1571170096577&impolicy=massimodutti-itxmediumhigh&imwidth=600',
+      color: 'green',
+      size: 12,
+      price: 60,
+      inventory: 7
+    }
+  ];
+
+  // these products are hard-coded here, as soon as wishlist is merged, products will come from the store. //
+
+  const deliveryCost = 35;
+  const voucher = 5;
+
+  const checkIfExists = (item) => userCart.find((cartItem) => item.id === cartItem.id);
+  const findItemPrice = (id) => products.find((item) => item.id === id).price;
+  const countTotal = userCart.reduce((acc, nextValue) => (acc + findItemPrice(nextValue.id) * nextValue.quantity), 0);
+  const renderTableHeader = () => cartTableHeadings.map((heading) => <th key={heading}>{heading}</th>);
+
+  const renderProduct = (item) => (
+    <CartItem
+      key={item.title}
+      item={item}
+      increment={incrementQuantity}
+      decrement={decrementQuantity}
+      userCart={userCart}
+      removeItemFromCart={removeItemFromCart}
+    />
+  );
+
+  const renderTable = () => (
+    <table className={`${CN}__table content`}>
+      <tbody>
+        <tr className={`${CN}__heading`}>{renderTableHeader()}</tr>
+        {products.map((item) => checkIfExists(item) && renderProduct(item))}
+      </tbody>
+    </table>
+  );
+
+  const renderNoItemsInCart = () => (
+    <div className={`${CN}__empty-page`}>
+      <h2 className={`${CN}__subtitle`}>No items in your cart!</h2>
+      <img
+        className={`${CN}__image col-6 col-center"`}
+        alt="no cart items"
+        src="src/assets/img/content/img-no-cartitems.png"
+      />
+    </div>
+  );
+
+  return (
+    <div className={cx(CN, className)}>
+      <Link className={`${CN}__link`} to="/products">
+        <div className={`${CN}__offer`}>
+          <h2>
+          Scoring a good fashion deal is as exhilarating as winning the lottery!
+          Try out your luck!
+          </h2>
+          <div
+            className={`col-6 col-center ${CN}__pseudo-button`}
+          >
+            Continue shopping
+          </div>
+        </div>
+      </Link>
+
+      <div className={`${CN} content`}>
+        <h1 className={`${CN}__title`}>Cart</h1>
+      </div>
+      <div className={`${CN} content`}>
+        {userCart.length ? renderTable(userCart)
```

Comment:
`renderTable()`!

Reply:
?

Reply:
If you look, u pass unnecessary parameter inside **renderTable**